### PR TITLE
etcdctl/command: print more details about ErrNoEndpoint

### LINKED ISSUE #3402: `client: no endpoints available` when using etcdctl
Start 3-member etcd cluster, and set `-listen-peer-urls` to some value unrelated, so etcd cluster cannot talk to each other.

```
$./bin/etcdctl -C 127.0.0.1:4001 --debug cluster-health
Error:  client: no endpoints available
```

There is no more details about the failure, which is hard to explain and reason about. Actually, this fails because it fails to sync when creating client struct. My thought is to print out the fact that it is syncing and curl command when Sync members.

/cc @mischief

### PR DESCRIPTION
**Pull request by yichengq**
This commit prints more details if getting ErrNoEndpoint when sync with
cluster. For cluster-health, it skips sync and anaylzes member one by one.

Before:
```
Error:  client: no endpoints available
```

After when normal operations:
```
etcd cluster has no published client endpoints.
Try '--no-sync' if you want to access non-published client endpoints(http://127.0.0.1:22379).
Error:  client: no endpoints available
```

After when cluster-health:
```
member 8211f1d0f64f3269 is unreachable: it has zero client endpoints available. Please ensure the member has published its advertise client urls.
member 91bc3c398fb3c146 is unreachable: it has zero client endpoints available. Please ensure the member has published its advertise client urls.
member fd422379fda50e48 is unreachable: it has zero client endpoints available. Please ensure the member has published its advertise client urls.
cluster is unhealthy
```

from #3402

### CODE REVIEW

File: etcdctl/command/util.go [hunk 1]
```diff
@@ -221,6 +221,10 @@ func mustNewClient(c *cli.Context) client.Client {
 		err := hc.Sync(ctx)
 		cancel()
 		if err != nil {
+			if err == client.ErrNoEndpoints {
+				fmt.Fprintf(os.Stderr, "Failed to get any client endpoints when synchronizing through %s.\n", strings.Join(eps, ","))
+				fmt.Fprintf(os.Stderr, "Please ensure that etcd members can talk to each other through advertise peer urls, and each member sets its advertise client urls non-empty.\n")
```
Comment (xiang90) [inline etcdctl/command/util.go]: can we remove this log line? I do not think line provides any useful information.

File: etcdctl/command/util.go [hunk 2]
```diff
@@ -221,6 +221,10 @@ func mustNewClient(c *cli.Context) client.Client {
 		err := hc.Sync(ctx)
 		cancel()
 		if err != nil {
+			if err == client.ErrNoEndpoints {
+				fmt.Fprintf(os.Stderr, "Failed to get any client endpoints when synchronizing through %s.\n", strings.Join(eps, ","))
```
Comment (xiang90) [inline etcdctl/command/util.go]: "Failed to sync cluster with endpoints [%v]". Ideally, we can print out the errors.

**Comment by yichengq** [comment]
Remove the resolve part because etcdctl targets cluster users instead of cluster deployer. Rewrite the sentense about the reason.

File: etcdctl/command/util.go [hunk 3]
```diff
@@ -221,6 +221,9 @@ func mustNewClient(c *cli.Context) client.Client {
 		err := hc.Sync(ctx)
 		cancel()
 		if err != nil {
+			if err == client.ErrNoEndpoints {
+				fmt.Fprintf(os.Stderr, "Got empty client urls from member list when synchronizing through %s.\n", strings.Join(eps, ","))
```
Comment (xiang90) [inline etcdctl/command/util.go]: etcd cluster has no published client endpoints.
Try `--no-sync` if you want to access a non-published(%v) client endpoint.

File: etcdctl/command/cluster_health.go [hunk 4]
```diff
@@ -44,7 +45,8 @@ func handleClusterHealth(c *cli.Context) {
 		Transport: tr,
 	}
 
-	mi := mustNewMembersAPI(c)
```
Comment (xiang90) [inline etcdctl/command/cluster_health.go]: I think we can remove the newly added arg and set no-sync flag explicitly.

Comment (yichengq) [inline etcdctl/command/cluster_health.go]: The environment is a codegangsta/cli.Context, and I don't find the way to set a flag. It looks read-only.

Comment (xiang90) [inline etcdctl/command/cluster_health.go]: how about add a new function called mustNewClientNoSync

Comment (yichengq) [inline etcdctl/command/cluster_health.go]: will try

**Comment by yichengq** [comment]
I have added mustNewClientNoSync and extract newClient as the underlying shared function. mustNewClientNoSync is almost the same with mustNewClient except that it doesn't have the code to do the sync.

File: etcdctl/command/cluster_health.go [hunk 5]
```diff
@@ -54,6 +56,11 @@ func handleClusterHealth(c *cli.Context) {
 	for {
 		health := false
 		for _, m := range ms {
+			if len(m.ClientURLs) == 0 {
+				fmt.Printf("member %s is unreachable: it has zero client endpoints available. Please ensure the member has published its advertise client urls.\n", m.ID)
```
Comment (xiang90) [inline etcdctl/command/cluster_health.go]: member %s is unreachable: no available published client urls.

**Comment by xiang90** [comment]
LGTM

**Pull request merged**